src/renderer/reader/components/TextSelectionPopup.tsx

```diff
@@ -0,0 +1,189 @@
+// Licensed to the Readium Foundation under one or more contributor license agreements.
+// Use of this source code is governed by a BSD-style license
+// that can be found in the LICENSE file exposed on Github (readium) in the project repository.
+// ==LICENSE-END==
+
+import * as React from "react";
+import * as Popover from "@radix-ui/react-popover";
+import { connect } from "react-redux";
+import { IReaderRootState } from "readium-desktop/common/redux/states/renderer/readerRootState";
+import { AnnotationEdit } from "./AnnotationEdit";
+import { IColor } from "@r2-navigator-js/electron/common/highlight";
+import { TDrawType } from "readium-desktop/common/redux/states/renderer/note";
+import { TDispatch } from "readium-desktop/typings/redux";
+import { readerLocalActionAnnotations } from "../redux/actions";
+
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+interface IBaseProps {
+}
+
+// IProps may typically extend:
+// RouteComponentProps
+// ReturnType<typeof mapStateToProps>
+// ReturnType<typeof mapDispatchToProps>
+interface IProps extends IBaseProps, ReturnType<typeof mapStateToProps>, ReturnType<typeof mapDispatchToProps> {
+}
+
+interface IState {
+    isOpen: boolean;
+    position: { x: number; y: number } | null;
+}
+
+class TextSelectionPopup extends React.Component<IProps, IState> {
+    private previousSelectionText: string | null = null;
+
+    constructor(props: IProps) {
+        super(props);
+        this.state = {
+            isOpen: false,
+            position: null,
+        };
+    }    public componentDidUpdate(prevProps: IProps) {
+        // Check if there's a new text selection
+        const currentSelection = this.props.selectionInfo;
+        const hadSelection = prevProps.selectionInfo?.cleanText;
+        const hasSelection = currentSelection?.cleanText;
+        const selectionIsNew = this.props.selectionIsNew;
+
+        console.log('TextSelectionPopup componentDidUpdate:', {
+            hasSelection: !!hasSelection,
+            hadSelection: !!hadSelection,
+            selectionIsNew,
+            currentText: hasSelection?.slice(0, 50),
+            previousText: this.previousSelectionText?.slice(0, 50),
+        });
+
+        // If we have a new selection that's different from the previous one
+        if (hasSelection && hasSelection !== this.previousSelectionText && selectionIsNew) {
+            console.log('Showing popup for new selection:', hasSelection.slice(0, 50));
+            this.previousSelectionText = hasSelection;
+            this.showPopup();
+        } else if (!hasSelection && hadSelection) {
+            // Selection was cleared
+            console.log('Hiding popup - selection cleared');
+            this.previousSelectionText = null;
+            this.hidePopup();
+        }
+    }    private showPopup = () => {
+        console.log('showPopup called');
+        
+        // Try to get position from current DOM selection first
```

Comment:
You cannot get the current DOM position of the selection from the reader, host of the navigator webview: 

see this: https://github.com/edrlab/r2-navigator-js/blob/48eea2f8bc8f00f666b32506c32c2bbdfb8f9bda/src/electron/renderer/dom.ts#L295